diff -urN mksh.org/check.t mksh/check.t
--- mksh.org/check.t	2015-07-10 21:36:57.000000000 +0200
+++ mksh/check.t	2015-07-13 09:29:23.388948055 +0200
@@ -7467,7 +7467,6 @@
 	nohup='nohup '
 	r='\builtin fc -e -'
 	source='PATH=$PATH:. \command .'
-	stop='\kill -STOP'
 	type='\builtin whence -v'
 ---
 name: aliases-1-hartz4
@@ -7532,7 +7531,6 @@
 	nohup='nohup '
 	r='\builtin fc -e -'
 	source='PATH=$PATH:. \command .'
-	stop='\kill -STOP'
 	type='\builtin whence -v'
 ---
 name: aliases-3b
@@ -7555,7 +7553,6 @@
 	nohup='nohup '
 	r='\builtin fc -e -'
 	source='PATH=$PATH:. \command .'
-	stop='\kill -STOP'
 	type='\builtin whence -v'
 ---
 name: aliases-2b-hartz4
diff -urN mksh.org/main.c mksh/main.c
--- mksh.org/main.c	2015-07-10 21:37:01.000000000 +0200
+++ mksh/main.c	2015-07-13 09:29:08.398575771 +0200
@@ -74,7 +74,6 @@
 #if !defined(ANDROID) && !defined(MKSH_UNEMPLOYED)
 	/* not in Android for political reasons */
 	/* not in ARGE mksh due to no job control */
-	"stop=\\kill -STOP",
 #endif
 	"autoload=\\typeset -fu",
 	"functions=\\typeset -f",
diff -urN mksh.org/mksh.1 mksh/mksh.1
--- mksh.org/mksh.1	2015-07-10 21:36:05.000000000 +0200
+++ mksh/mksh.1	2015-07-13 09:29:15.412083285 +0200
@@ -1131,7 +1131,6 @@
 nohup=\*(aqnohup \*(aq
 r=\*(aq\ebuiltin fc \-e \-\*(aq
 source=\*(aqPATH=$PATH:. \ecommand .\*(aq
-stop=\*(aq\ekill \-STOP\*(aq
 type=\*(aq\ebuiltin whence \-v\*(aq
 .Ed
 .Pp
