--- mksh-42/sh.h~	2013-02-12 07:20:52.764918024 +0100
+++ mksh-42/sh.h	2013-02-12 07:21:57.486460200 +0100
@@ -166,7 +166,7 @@
 #ifdef EXTERN
 __RCSID("$MirOS: src/bin/mksh/sh.h,v 1.634.2.1 2013/02/11 17:25:03 tg Exp $");
 #endif
-#define MKSH_VERSION "R42 2013/02/10"
+#define MKSH_VERSION "R42 2013/02/10 @DISTRO@"
 
 /* arithmetic types: C implementation */
 #if !HAVE_CAN_INTTYPES
--- mksh-42/check.t~	2013-02-11 14:07:19.000000000 +0100
+++ mksh-42/check.t	2013-02-12 07:21:38.082664463 +0100
@@ -29,7 +29,7 @@
 # http://www.freebsd.org/cgi/cvsweb.cgi/src/tools/regression/bin/test/regress.sh?rev=HEAD
 
 expected-stdout:
-	@(#)MIRBSD KSH R42 2013/02/10
+	@(#)MIRBSD KSH R42 2013/02/10 @DISTRO@
 description:
 	Check version of shell.
 stdin:
