--- mksh-50/main.c.orig	2014-08-21 18:13:38.072682886 +0200
+++ mksh-50/main.c	2014-08-21 18:29:13.715976954 +0200
@@ -591,9 +591,14 @@
 		if (Flag(FLOGIN))
 			include(substitute("$HOME/.profile", 0), 0, NULL, true);
 		if (Flag(FTALKING)) {
-			cp = substitute("${ENV:-" MKSHRC_PATH "}", DOTILDE);
+			cp = substitute("${ENV}", DOTILDE);
 			if (cp[0] != '\0')
 				include(cp, 0, NULL, true);
+			else {
+				cp = substitute(MKSHRC_PATH, DOTILDE);
+				include("/etc/mkshrc", 0, NULL, true);
+				include(cp, 0, NULL, true);
+			}
 		}
 	} else {
 		include(MKSH_SUID_PROFILE, 0, NULL, true);
